### ISSUE TITLE
Redirect skips bypass_auth value

### ISSUE THREAD

**Issue by Paszymaja** [emissary-ingress/emissary#4419]
**Describe the bug**
Hi, recently we moved from ambassador to emissary and one of our redirect mappings stopped working. It should redirect from /test to /test/. It looks like the `bypass_auth: true` value is skipped, the request is still falidated and it returns a 401 error. If we log in the redirect works correctly.

**To Reproduce**
1. With configured auth service deploy:
```
apiVersion: getambassador.io/v3alpha1
kind: Mapping
metadata:
  name: redirect
spec:
  hostname: '*'
  service: "adress.com"
  prefix: /test
  bypass_auth: true
  host_redirect: true
  prefix_regex: true
  path_redirect: /test/
---
apiVersion: getambassador.io/v3alpha1
kind: Mapping
metadata:
  name: test-mapping
spec:
  hostname: 'adress.com'
  service: "service:8080"
  prefix: /test/
  bypass_auth: true
```
2. When going to /test get the 401 error even though bypass_auth is set to true.

**Expected behavior**
`bypass_auth: true` should not be skipped and the request should be redirected unauthorised from /test to /test/

**Versions (please complete the following information):**
 - Emissary: 3.1.0
 - Kubernetes environment Google Kubernetes Engine
 - Version: 1.23.6

**Comment by teejaded** [comment]
https://github.com/emissary-ingress/emissary/issues/4640

**Comment by mehighlow** [comment]
For time travelers,

In Emissary-ingress below v3.3.0, when an AuthService is applied auth happens before performing the host_redirect.

E.g.
In case you have a domain.com to www.domain.com redirect(`host_redirect: true`) and auth service installed (is not applied to domain.com, but to any other domain), then requests to http(s)://domain.com will rise authentication but http(s):www.domain.com won't.

This has been fixed with https://github.com/emissary-ingress/emissary/pull/4641 and released in v3.3.0

I think this issue may be closed.
